TBTK/calc/TightBindingLib/src/FileReader.cpp

```diff
@@ -892,6 +895,245 @@ void FileReader::readAttributes(
+//      printf ("\n");
+//    }
+//
+//    ret = H5Tclose(atype);
+//    ret = H5Sclose(aspace);
+//    ret = H5Aclose(attr);
+//
+//    return 0;
+//}
+
+
+Util::ParameterSet* FileReader::readParameterSet(
+    string name,
+	string path
+){
+	try{
+		stringstream ss;
+		ss << path;
+		if(path.back() != '/')
+			ss << "/";
+		ss << name;
+
+		Exception::dontPrint();
+
+        Util::ParameterSet *ps = new Util::ParameterSet();
+
+        H5File file(filename, H5F_ACC_RDONLY);
+		DataSet dataset = file.openDataSet(name + "Int");
+        unsigned int num = dataset.getNumAttrs();
+
+		for(unsigned int n = 0; n < num; n++){
+			Attribute attribute = dataset.openAttribute(n);
+			DataType type = attribute.getDataType();
+            string nameAttribute;
+			nameAttribute = attribute.getName();
+
+			TBTKAssert(
+				type == PredType::STD_I64BE,
+				"FileReader::readParameterSet()",
+				"The attribute '" << nameAttribute << "' is not of integer type.",
+				""
+			);
+			int value;
+			attribute.read(PredType::NATIVE_INT, &value);
+			ps->addInt(nameAttribute, value);
+		}
+
+        dataset = file.openDataSet(name + "Double");
+        num = dataset.getNumAttrs();
+
+		for(unsigned int n = 0; n < num; n++){
+			Attribute attribute = dataset.openAttribute(n);
+			DataType type = attribute.getDataType();
+            string nameAttribute;
+			nameAttribute = attribute.getName();
+
+			TBTKAssert(
+				type == PredType::IEEE_F64BE,
+				"FileReader::readParameterSet()",
+				"The attribute '" << nameAttribute << "' is not of double type.",
+				""
+			);
+			double value;
+			attribute.read(PredType::NATIVE_DOUBLE, &value);
+			ps->addDouble(nameAttribute, value);
+		}
+
+        dataset = file.openDataSet(name + "Complex");
+        num = dataset.getNumAttrs();
+        const complex<double> one(1,0);
+        const complex<double> i(0,1);
+
+		for(unsigned int n = 0; n < num; n++){
+			Attribute attribute = dataset.openAttribute(n);
+			DataType type = attribute.getDataType();
+            string nameAttribute;
+			nameAttribute = attribute.getName();
+			const hsize_t dimension = 2;
+			ArrayType dataTypeComplex(PredType::NATIVE_DOUBLE, 1, &dimension);
+
+			TBTKAssert(
+				type == dataTypeComplex,
+				"FileReader::readParameterSet()",
+				"The attribute '" << nameAttribute << "' is not of complex type.",
+				""
+			);
+			double value[2];
+			attribute.read(dataTypeComplex, value);
+            complex<double> complexValue = value[0]*one;
+            complexValue += value[1]*i;
+            ps->addComplex(nameAttribute, complexValue);
+		}
+
+        dataset = file.openDataSet(name + "String");
+        num = dataset.getNumAttrs();
+
+		for(unsigned int n = 0; n < num; n++){
+			Attribute attribute = dataset.openAttribute(n);
+			DataType type = attribute.getDataType();
+            string nameAttribute = attribute.getName();
+			unsigned int memLength = attribute.getInMemDataSize();
+			StrType strDataType(PredType::C_S1, memLength);
```

Comment:
Make sure this name convention agrees with that for complexType/complexDataType above (currently dataTypeComplex).